runtime/jvmti/jvmtiHook.c

```diff
@@ -1071,6 +1070,16 @@ jvmtiHookFieldModification(J9HookInterface** hook, UDATA eventNum, void* eventDa
 				if (methodID != NULL) {
 					callback((jvmtiEnv *) j9env, (JNIEnv *) currentThread, threadRef, methodID, (jlocation) location, (jclass) classRef, (jobject) objectRef, fieldID, signatureType, newValue);
 				}
+				vm->internalVMFunctions->internalEnterVMFromJNI(currentThread);
+				if (NULL != object) {
+					J9VMPutFieldEvent* data = eventData;
+					data->object = *objectRef;
```

Comment:
... And this is kept alive because it's been pushed on the stack?  What kind of frame is there that arg0EA -1 is guaranteed to have space and not overwrite something else?  Also, shouldn't we have to adjust a count?

I realize that's all existing code but some clarifications would help

Reply:
Yes, it is pushed into the arg area of the frame created by pushEventFrame (called from prepareForEvent).  The frame is a J9SFJNINativeMethodFrame with a NULL method - the stack walker treats all slots between arg0EA and the frame to be JNI local ref object pushes.